tests.lua

```diff
@@ -217,6 +241,26 @@ function LibSerialize:RunTests()
         elseif typ ~= "table" and value ~= deserialized then
             fail(index, fromVer, toVer, value, ("Non-matching deserialization result: %s"):format(tostring(deserialized)))
         end
+
+        -- Async tests
```

Comment:
This is better, but you're still not testing any scenarios that actually yield.

Reply:
It's now checking the deserialized vs original values. The `nil` and `0/0` tests are still failing the bytelen tests for async serialize for some reason I can't figure out, but in both cases they do deserialize correctly.

Reply:
Maybe my comments in the deserialization code will fix it?

Reply:
No updates, still not sure why async would return a different bytelen here.